Compiler crashes during build on ALCF systems

```
[  5%] Built target ocml_host_math_funcs
[  7%] Built target LLVMHipPasses
[  7%] Built target irif
[  7%] Built target prepare-builtins
[ 55%] Built target ocml
[ 55%] Built target oclc_correctly_rounded_sqrt_off
[ 55%] Built target oclc_daz_opt_off
[ 55%] Built target oclc_finite_only_off
[ 57%] Built target oclc_isa_version_803
[ 60%] Built target oclc_unsafe_math_off
[ 60%] Built target devicelib_bc
[ 63%] Built target rtdevlib
[ 65%] Built target hipcc.bin
[ 65%] Built target hipconfig.bin
[ 65%] Built target CHIP
[ 65%] Generating hipInfo
/home/pvelesko/chipStar/samples/hipInfo/hipInfo.cpp:146:5: warning: ignoring return value of function declared with 'nodiscard' attribute [-Wunused-result]
  146 |     hipGetDeviceCount(&deviceCnt);
      |     ^~~~~~~~~~~~~~~~~ ~~~~~~~~~~
/home/pvelesko/chipStar/samples/hipInfo/hipInfo.cpp:150:9: warning: ignoring return value of function declared with 'nodiscard' attribute [-Wunused-result]
  150 |         hipDeviceCanAccessPeer(&isPeer, i, deviceId);
      |         ^~~~~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~
/home/pvelesko/chipStar/samples/hipInfo/hipInfo.cpp:159:9: warning: ignoring return value of function declared with 'nodiscard' attribute [-Wunused-result]
  159 |         hipDeviceCanAccessPeer(&isPeer, i, deviceId);
      |         ^~~~~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~
/home/pvelesko/chipStar/samples/hipInfo/hipInfo.cpp:180:5: warning: ignoring return value of function declared with 'nodiscard' attribute [-Wunused-result]
  180 |     hipMemGetInfo(&free, &total);
      |     ^~~~~~~~~~~~~ ~~~~~~~~~~~~~
4 warnings generated when compiling for .
/opt/aurora/24.086.0/spack/gcc/0.7.0/install/linux-sles15-x86_64/gcc-12.2.0/gcc-12.2.0-jf4ov3v3scg7dvd76qhsuugl3jp42gfn/lib/gcc/x86_64-pc-linux-gnu/12.2.0/../../../../include/c++/12.2.0/optional:484: constexpr const _Tp& std::_Optional_base_impl<_Tp, _Dp>::_M_get() const [with _Tp = bool; _Dp = std::_Optional_base<bool, true, true>]: Assertion 'this->_M_is_engaged()' failed.
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
0.	Program arguments: /home/pvelesko/install/llvm/19.0/bin/opt /var/tmp/pbs.792471.aurora-pbs-0001.hostmgmt.cm.aurora.alcf.anl.gov/hipInfo-generic-b0c799-link-9f3ec8.bc -load-pass-plugin /home/pvelesko/chipStar/build/lib/libLLVMHipSpvPasses.so -passes=hip-post-link-passes -o /var/tmp/pbs.792471.aurora-pbs-0001.hostmgmt.cm.aurora.alcf.anl.gov/hipInfo-generic-b0c799-lower-feaa36.bc
1.	Running pass "cgscc(inline<only-mandatory>,inline),cgscc()" on module "/var/tmp/pbs.792471.aurora-pbs-0001.hostmgmt.cm.aurora.alcf.anl.gov/hipInfo-generic-b0c799-link-9f3ec8.bc"
2.	Running pass "cgscc(inline<only-mandatory>,inline)" on module "/var/tmp/pbs.792471.aurora-pbs-0001.hostmgmt.cm.aurora.alcf.anl.gov/hipInfo-generic-b0c799-link-9f3ec8.bc"
 #0 0x00000000033aa0ab llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/home/pvelesko/install/llvm/19.0/bin/opt+0x33aa0ab)
 #1 0x00000000033a74eb llvm::sys::RunSignalHandlers() (/home/pvelesko/install/llvm/19.0/bin/opt+0x33a74eb)
 #2 0x00000000033a7615 SignalHandler(int) Signals.cpp:0:0
 #3 0x00001526c56ee8c0 __restore_rt (/lib64/libpthread.so.0+0x168c0)
 #4 0x00001526c4e46c6b raise (/lib64/libc.so.6+0x4ac6b)
 #5 0x00001526c4e48305 abort (/lib64/libc.so.6+0x4c305)
 #6 0x00001526c5234d8e (/opt/aurora/24.086.0/spack/gcc/0.7.0/install/linux-sles15-x86_64/gcc-12.2.0/gcc-12.2.0-jf4ov3v3scg7dvd76qhsuugl3jp42gfn/lib64/libstdc++.so.6+0xd8d8e)
 #7 0x00000000028daeef (anonymous namespace)::InlineCostCallAnalyzer::InlineCostCallAnalyzer(llvm::Function&, llvm::CallBase&, llvm::InlineParams const&, llvm::TargetTransformInfo const&, llvm::function_ref<llvm::AssumptionCache& (llvm::Function&)>, llvm::function_ref<llvm::BlockFrequencyInfo& (llvm::Function&)>, llvm::ProfileSummaryInfo*, llvm::OptimizationRemarkEmitter*, bool, bool) InlineCost.cpp:0:0
 #8 0x00000000028e9757 llvm::getInlineCost(llvm::CallBase&, llvm::Function*, llvm::InlineParams const&, llvm::TargetTransformInfo&, llvm::function_ref<llvm::AssumptionCache& (llvm::Function&)>, llvm::function_ref<llvm::TargetLibraryInfo const& (llvm::Function&)>, llvm::function_ref<llvm::BlockFrequencyInfo& (llvm::Function&)>, llvm::ProfileSummaryInfo*, llvm::OptimizationRemarkEmitter*) (/home/pvelesko/install/llvm/19.0/bin/opt+0x28e9757)
 #9 0x00000000028e9c8e llvm::getInlineCost(llvm::CallBase&, llvm::InlineParams const&, llvm::TargetTransformInfo&, llvm::function_ref<llvm::AssumptionCache& (llvm::Function&)>, llvm::function_ref<llvm::TargetLibraryInfo const& (llvm::Function&)>, llvm::function_ref<llvm::BlockFrequencyInfo& (llvm::Function&)>, llvm::ProfileSummaryInfo*, llvm::OptimizationRemarkEmitter*) (/home/pvelesko/install/llvm/19.0/bin/opt+0x28e9c8e)
#10 0x00000000028eb9cb llvm::InlineCost llvm::function_ref<llvm::InlineCost (llvm::CallBase&)>::callback_fn<getDefaultInlineAdvice(llvm::CallBase&, llvm::AnalysisManager<llvm::Function>&, llvm::InlineParams const&)::'lambda2'(llvm::CallBase&)>(long, llvm::CallBase&) InlineAdvisor.cpp:0:0
#11 0x00000000028efcbc llvm::shouldInline(llvm::CallBase&, llvm::function_ref<llvm::InlineCost (llvm::CallBase&)>, llvm::OptimizationRemarkEmitter&, bool) (/home/pvelesko/install/llvm/19.0/bin/opt+0x28efcbc)
#12 0x00000000028f1858 getDefaultInlineAdvice(llvm::CallBase&, llvm::AnalysisManager<llvm::Function>&, llvm::InlineParams const&) InlineAdvisor.cpp:0:0
#13 0x00000000028f1a0b llvm::DefaultInlineAdvisor::getAdviceImpl(llvm::CallBase&) (/home/pvelesko/install/llvm/19.0/bin/opt+0x28f1a0b)
#14 0x00000000028edccd llvm::InlineAdvisor::getAdvice(llvm::CallBase&, bool) (/home/pvelesko/install/llvm/19.0/bin/opt+0x28edccd)
#15 0x0000000001bb1df4 llvm::InlinerPass::run(llvm::LazyCallGraph::SCC&, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>&, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&) (/home/pvelesko/install/llvm/19.0/bin/opt+0x1bb1df4)
#16 0x0000000001275acd llvm::detail::PassModel<llvm::LazyCallGraph::SCC, llvm::InlinerPass, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&>::run(llvm::LazyCallGraph::SCC&, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>&, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&) (/home/pvelesko/install/llvm/19.0/bin/opt+0x1275acd)
#17 0x000000000280a754 llvm::PassManager<llvm::LazyCallGraph::SCC, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&>::run(llvm::LazyCallGraph::SCC&, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>&, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&) (/home/pvelesko/install/llvm/19.0/bin/opt+0x280a754)
#18 0x000000000127468d llvm::detail::PassModel<llvm::LazyCallGraph::SCC, llvm::PassManager<llvm::LazyCallGraph::SCC, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&>, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&>::run(llvm::LazyCallGraph::SCC&, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>&, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&) (/home/pvelesko/install/llvm/19.0/bin/opt+0x127468d)
#19 0x000000000280c149 llvm::ModuleToPostOrderCGSCCPassAdaptor::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/home/pvelesko/install/llvm/19.0/bin/opt+0x280c149)
#20 0x000000000127466d llvm::detail::PassModel<llvm::Module, llvm::ModuleToPostOrderCGSCCPassAdaptor, llvm::AnalysisManager<llvm::Module>>::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/home/pvelesko/install/llvm/19.0/bin/opt+0x127466d)
#21 0x00000000031c3584 llvm::PassManager<llvm::Module, llvm::AnalysisManager<llvm::Module>>::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/home/pvelesko/install/llvm/19.0/bin/opt+0x31c3584)
#22 0x0000000001bb0053 llvm::ModuleInlinerWrapperPass::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/home/pvelesko/install/llvm/19.0/bin/opt+0x1bb0053)
#23 0x000000000127424d llvm::detail::PassModel<llvm::Module, llvm::ModuleInlinerWrapperPass, llvm::AnalysisManager<llvm::Module>>::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/home/pvelesko/install/llvm/19.0/bin/opt+0x127424d)
#24 0x00000000031c3584 llvm::PassManager<llvm::Module, llvm::AnalysisManager<llvm::Module>>::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/home/pvelesko/install/llvm/19.0/bin/opt+0x31c3584)
#25 0x00000000007ebb3d llvm::runPassPipeline(llvm::StringRef, llvm::Module&, llvm::TargetMachine*, llvm::TargetLibraryInfoImpl*, llvm::ToolOutputFile*, llvm::ToolOutputFile*, llvm::ToolOutputFile*, llvm::StringRef, llvm::ArrayRef<llvm::PassPlugin>, llvm::ArrayRef<std::function<void (llvm::PassBuilder&)>>, llvm::opt_tool::OutputKind, llvm::opt_tool::VerifierKind, bool, bool, bool, bool, bool, bool, bool) (/home/pvelesko/install/llvm/19.0/bin/opt+0x7ebb3d)
#26 0x00000000007ddecc optMain (/home/pvelesko/install/llvm/19.0/bin/opt+0x7ddecc)
#27 0x00001526c4e3124d __libc_start_main (/lib64/libc.so.6+0x3524d)
#28 0x00000000007d421a _start /home/abuild/rpmbuild/BUILD/glibc-2.31/csu/../sysdeps/x86_64/start.S:122:0
clang++: error: unable to execute command: Aborted
clang++: error: hipspv-link command failed due to signal (use -v to see invocation)
clang version 19.1.0-rc1 (https://github.com/CHIP-SPV/llvm-project.git 9183abab088b12dadd819f7dc096741f40c8218e)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/pvelesko/install/llvm/19.0/bin
Build config: +assertions
clang++: note: diagnostic msg: Error generating preprocessed source(s).

failed to execute:/home/pvelesko/install/llvm/19.0/bin/clang++ -include /home/pvelesko/chipStar/include/hip/spirv_fixups.h -I//home/pvelesko/chipStar/build/include -x hip /home/pvelesko/chipStar/samples/hipInfo/hipInfo.cpp -D__HIP_PLATFORM_SPIRV__= --offload=spirv64 -nohipwrapperinc --hip-path=/home/pvelesko/chipStar/build --target=x86_64-unknown-linux-gnu   -I/home/pvelesko/chipStar/include -I/home/pvelesko/chipStar/HIP/include -I/home/pvelesko/chipStar/build/include -o /home/pvelesko/chipStar/build/hipInfo -L/home/pvelesko/chipStar/build -lCHIP -no-hip-rt -locml_host_math_funcs -Wl,-rpath,/home/pvelesko/chipStar/build
make[2]: *** [CMakeFiles/hipInfoBin.dir/build.make:75: hipInfo] Error 1
make[1]: *** [CMakeFiles/Makefile2:2984: CMakeFiles/hipInfoBin.dir/all] Error 2
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,6 +23,13 @@
 #
 # =============================================================================
 
+execute_process(
+  COMMAND uname -m
+  OUTPUT_VARIABLE DETECTED_ARCHITECTURE
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+message(STATUS "Detected architecture: ${DETECTED_ARCHITECTURE}")
+
 # temporary
 add_compile_options(-Wno-format-extra-args)
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,6 +30,21 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-duplicate-decl-specifier \
 -Wno-tautological-constant-compare  -Wno-c++20-extensions -Wno-unused-result \
 -Wno-delete-abstract-non-virtual-dtor -Wno-deprecated-declarations -Wunused-command-line-argument -gdwarf-4")
 
+# Find GCC installation path
+execute_process(
+  COMMAND which gcc
+  OUTPUT_VARIABLE GCC_PATH
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+
+# Extract the base path (removing /bin/gcc from the end)
+get_filename_component(GCC_BASE_PATH ${GCC_PATH} DIRECTORY)
+get_filename_component(GCC_BASE_PATH ${GCC_BASE_PATH} DIRECTORY)
+
+message(STATUS "GCC base path: ${GCC_BASE_PATH}")
+# Necessary on some systems with nonstandard GCC installations
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --gcc-toolchain=${GCC_BASE_PATH}")
+
 # end temporary
 cmake_minimum_required(VERSION 3.20 FATAL_ERROR)
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")

--- a/README.md
+++ b/README.md
@@ -217,6 +217,7 @@ CHIP_LOGLEVEL=<trace/debug/info/warn/err/crit>  # Sets the log level. If compile
 CHIP_DUMP_SPIRV=<ON/OFF(default)>               # Dumps the generated SPIR-V code to a file
 CHIP_JIT_FLAGS_OVERRIDE=<flags>                 # String to override the default JIT flags. Defaults to -cl-kernel-arg-info -cl-std=CL3.0
 CHIP_L0_COLLECT_EVENTS_TIMEOUT=<N(30s default)> # Timeout in seconds for collecting Level Zero events
+CHIP_L0_EVENT_TIMEOUT=<N(0 default)             # Timeout in seconds for how long Level Zero should wait on an event before timing out
 CHIP_SKIP_UNINIT=<ON/OFF(default)>              # If enabled, skips the uninitialization of chipStar's backend objects at program termination
 ```
 

--- a/bin/cucc.py
+++ b/bin/cucc.py
@@ -26,13 +26,12 @@
 #=============================================================================
 """Compiler wrapper aiming to be a drop-in replacement for nvcc."""
 # Enable v3.9+ type annotations in older versions (down to 3.7).
-from __future__ import annotations
 import argparse
 import os
 import subprocess
 import sys
 from shlex import quote
-
+from typing import List, Set, Optional
 
 # If true, we change the behavior of this tool so that the CMake
 # considers this tool to be NVvidia's nvcc. This allows us to compile
@@ -165,7 +164,7 @@ def get_cuda_library_dir() -> str:
     """
     return get_hip_path() + "/lib"
 
-def determine_input_languages(arg_list : list[str], xmode=None) -> set[str]:
+def determine_input_languages(arg_list: List[str], xmode: Optional[str] = None) -> Set[str]:
     """Determine input language modes from the argument list
 
     The 'xmode' specifies value of the last '-x' option if it was specified.
@@ -201,7 +200,7 @@ def determine_input_languages(arg_list : list[str], xmode=None) -> set[str]:
 
     return modes
 
-def filter_args_for_hipcc(arg_list : list[str]) -> list[str]:
+def filter_args_for_hipcc(arg_list: List[str]) -> List[str]:
     """Filter out arguments on the way to hipcc.
     """
     filtered = []

--- a/cmake/FindLLVM.cmake
+++ b/cmake/FindLLVM.cmake
@@ -109,6 +109,11 @@ if(NOT DEFINED LLVM_SPIRV)
   if(NOT LLVM_SPIRV)
     message(FATAL_ERROR "Can't find llvm-spirv. Please provide CMake argument -DLLVM_SPIRV=/path/to/llvm-spirv<-version>")
   endif()
+else()
+  # Check if the provided LLVM_SPIRV file exists
+  if(NOT EXISTS ${LLVM_SPIRV})
+    message(FATAL_ERROR "Provided LLVM_SPIRV (${LLVM_SPIRV}) does not exist")
+  endif()
 endif()
 message(STATUS "Using llvm-spirv: ${LLVM_SPIRV}")
 

--- a/docker/DockerfileBase
+++ b/docker/DockerfileBase
@@ -1,4 +1,4 @@
-FROM ubuntu:latest
+FROM ubuntu:24.04
 
 ENV DEBIAN_FRONTEND noninteractive
 

--- a/docker/DockerfileBase
+++ b/docker/DockerfileBase
@@ -1,19 +1,22 @@
-FROM ubuntu:24.04
+FROM ubuntu:22.04
 
 ENV DEBIAN_FRONTEND noninteractive
 
 SHELL ["/bin/bash", "-ci"]
 
+# Set timezone non-interactively
+ENV TZ=Etc/UTC
+RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
+
 # Get the basic stuff and create chipStarUser user with sudo privileges
 RUN apt-get update && \
-    apt-get -y upgrade && \
-    apt-get install -y sudo; \
+    apt-get install -y sudo tzdata && \
     useradd -ms /bin/bash chipStarUser && \
     groupadd -f video && groupadd -f render && \
     usermod -aG sudo,video,render chipStarUser; \
     echo '%sudo ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers
 
-# Set as default user
+
 USER chipStarUser
 WORKDIR /home/chipStarUser
 
@@ -113,7 +116,8 @@ RUN wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -
 # Set up conda environment
 ENV PATH="/apps/conda/bin:${PATH}"
 RUN conda init bash && \
-    conda config --add channels intel && \
+    conda config --add channels https://software.repos.intel.com/python/conda/ && \
+    conda config --add channels conda-forge && \
     conda create -n oneapi-2024.1.0 -y
 
 # Install OneAPI packages
@@ -139,24 +143,36 @@ RUN conda run -n oneapi-2024.1.0 conda install -y \
     onemkl-sycl-sparse=2024.1.0=intel_691 \
     onemkl-sycl-stats=2024.1.0=intel_691 \
     onemkl-sycl-vm=2024.1.0=intel_691 \
-    tbb=2021.12.0=intel_495 \
-    level-zero=1.17.6
+    tbb=2021.12.0=intel_495
 
-# RUN conda config --add channels conda-forge && conda run -n oneapi-2024.1.0 conda install -y intel-compute-runtime
-RUN sudo apt install -y intel-opencl-icd vim
+USER root
+# Install level-zero and other dependencies
+RUN sudo apt-get update && \
+    sudo apt-get install -y intel-opencl-icd vim
 
-# Install Level Zero headers system-wide
-RUN mkdir -p /tmp/level-zero-headers && \
-    cd /tmp/level-zero-headers && \
-    git clone https://github.com/oneapi-src/level-zero.git && \
+# Install dependencies for building Level Zero
+RUN apt-get update && apt-get install -y \
+    cmake \
+    pkg-config \
+    build-essential \
+    git
+
+# Clone, build, and install Level Zero
+RUN git clone https://github.com/oneapi-src/level-zero.git && \
     cd level-zero && \
     mkdir build && \
     cd build && \
     cmake .. && \
-    make -j && \
-    sudo make install && \
-    cd ~/ && \
-    rm -rf /tmp/level-zero-headers
+    make -j$(nproc) && \
+    make install && \
+    cd ../.. && \
+    rm -rf level-zero
+
+# Update library cache
+RUN ldconfig
+
+# Optional: If you encounter solver issues, set the following environment variable
+ENV CONDA_LIBMAMBA_SOLVER_NO_CHANNELS_FROM_INSTALLED=1
 
 # Delete Intel OpenCL Loader
 RUN rm -f /apps/conda/envs/oneapi-2024.1.0/lib/libOpenCL.so.1
@@ -179,8 +195,16 @@ RUN mkdir -p /apps/modulefiles/Core/oneapi && \
     echo 'setenv("ONEAPI_ROOT", base)' >> /apps/modulefiles/Core/oneapi/2024.1.0.lua && \
     echo 'setenv("ONEAPI_VERSION", version)' >> /apps/modulefiles/Core/oneapi/2024.1.0.lua
 
-# Set permissionst to enable non-root access to GPUs
+RUN pip install pyyaml
+
+RUN apt-get update && apt-get install -y vim-common # for xxd
+
+# Set permissions to enable non-root access to GPUs
 RUN echo 'sudo chmod 666 /dev/dri/* 2>/dev/null || true' >> /home/chipStarUser/.bashrc
+
+# Set as default user
+USER chipStarUser
+WORKDIR /home/chipStarUser
 
 #////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 #//                                                                                                                                        //

--- a/docker/DockerfileLatest
+++ b/docker/DockerfileLatest
@@ -14,7 +14,7 @@ RUN clinfo -l
 
 RUN pip install pyyaml
 
-RUN sudo apt-get install vim-common -y # for xxd
+RUN sudo apt update && sudo apt install vim-common -y # for xxd
 
 ENV CHIP_BE=opencl
 ENV CHIP_DEVICE_TYPE=cpu

--- a/docker/DockerfileLatest
+++ b/docker/DockerfileLatest
@@ -12,10 +12,6 @@ SHELL ["/bin/bash", "-ci"]
 
 RUN clinfo -l
 
-RUN pip install pyyaml
-
-RUN apt-get update && apt-get install -y vim-common # for xxd
-
 ENV CHIP_BE=opencl
 ENV CHIP_DEVICE_TYPE=cpu
 ENV CHIP_LOGLEVEL=info

--- a/docker/DockerfileLatest
+++ b/docker/DockerfileLatest
@@ -14,7 +14,7 @@ RUN clinfo -l
 
 RUN pip install pyyaml
 
-RUN sudo apt update && sudo apt install vim-common -y # for xxd
+RUN apt-get update && apt-get install -y vim-common # for xxd
 
 ENV CHIP_BE=opencl
 ENV CHIP_DEVICE_TYPE=cpu

--- a/samples/CMakeLists.txt
+++ b/samples/CMakeLists.txt
@@ -131,8 +131,11 @@ set(SAMPLES
     ccompat
     hipComplex
     hipHostMallocSample
-    hipDeviceLink
 )
+
+if (NOT "${DETECTED_ARCHITECTURE}" STREQUAL "riscv64")
+  list(APPEND SAMPLES hipDeviceLink)
+endif()
 
 include(mkl_and_icpx)
 # Add samples that depend on Intel's oneAPI compiler - if available.

--- a/samples/fp16/CMakeLists.txt
+++ b/samples/fp16/CMakeLists.txt
@@ -1,6 +1,7 @@
-
-add_chip_test(fp16 fp16 PASSED haxpy-base.cpp)
-
-add_chip_test(fp16_math fp16_math PASSED half_math.cpp)
-
-add_chip_test(fp16_half2_math fp16_half2_math PASSED half2_math.cpp)
+if(DETECTED_ARCHITECTURE STREQUAL "riscv64")
+  message(STATUS "RISC-V 64-bit architecture detected. Skipping fp16 tests.")
+else()
+  add_chip_test(fp16 fp16 PASSED haxpy-base.cpp)
+  add_chip_test(fp16_math fp16_math PASSED half_math.cpp)
+  add_chip_test(fp16_half2_math fp16_half2_math PASSED half2_math.cpp)
+endif()

--- a/scripts/configure_llvm.sh
+++ b/scripts/configure_llvm.sh
@@ -119,14 +119,24 @@ else
   cd ${LLVM_DIR}
 fi
 
-# check if the build directory exists, if not create it
-if [ ! -d build_$VERSION ]; then
+# check if the build directory exists
+if [ -d build_$VERSION ]; then
+  read -p "Build directory build_$VERSION already exists. Do you want to delete it and continue? (y/n) " answer
+  case ${answer:0:1} in
+    y|Y )
+      echo "Deleting existing build directory..."
+      rm -rf build_$VERSION
+      mkdir build_$VERSION
+      cd build_$VERSION
+    ;;
+    * )
+      echo "Build directory not deleted. Exiting."
+      exit 1
+    ;;
+  esac
+else
   mkdir build_$VERSION
   cd build_$VERSION
-else
-  # Warn the user, error out
-  echo "Build directory build_$VERSION already exists, please remove it and re-run the script"
-  exit 1
 fi
 
 # Check if /usr/include/plugin-api.h exists
@@ -143,33 +153,38 @@ elif [ -f /usr/include/plugin-api.h ]; then
   BINUTILS_HEADER_DIR=/usr/include
 else
   echo "plugin-api.h was not found at /usr/include/plugin-api.h"
-  echo "Installing binutils-dev from source..."
-
-  # Define the installation location
+  
+  # Check if binutils was installed in a previous attempt
   BINUTILS_INSTALL_DIR=${INSTALL_DIR}/binutils
-  
-  # Create the installation directory if it doesn't exist
-  mkdir -p ${BINUTILS_INSTALL_DIR}
-  
-  # Download the binutils source
-  BINUTILS_VERSION="2.36.1"
-  wget https://ftp.gnu.org/gnu/binutils/binutils-${BINUTILS_VERSION}.tar.gz
-  
-  # Extract the source
-  tar -xzf binutils-${BINUTILS_VERSION}.tar.gz
-  cd binutils-${BINUTILS_VERSION}
-  
-  # Configure, compile, and install binutils
-  ./configure --prefix=${BINUTILS_INSTALL_DIR}
-  make -j$(nproc)
-  make install
-  
-  # Clean up
-  cd ..
-  rm -rf binutils-${BINUTILS_VERSION} binutils-${BINUTILS_VERSION}.tar.gz
-  
-  echo "binutils-dev installed successfully in ${BINUTILS_INSTALL_DIR}"
-  BINUTILS_HEADER_DIR=${BINUTILS_INSTALL_DIR}/include
+  if [ -f "${BINUTILS_INSTALL_DIR}/include/plugin-api.h" ]; then
+    echo "Found previously installed binutils at ${BINUTILS_INSTALL_DIR}"
+    BINUTILS_HEADER_DIR=${BINUTILS_INSTALL_DIR}/include
+  else
+    echo "Installing binutils-dev from source..."
+    
+    # Create the installation directory if it doesn't exist
+    mkdir -p ${BINUTILS_INSTALL_DIR}
+    
+    # Download the binutils source
+    BINUTILS_VERSION="2.36.1"
+    wget https://ftp.gnu.org/gnu/binutils/binutils-${BINUTILS_VERSION}.tar.gz
+    
+    # Extract the source
+    tar -xzf binutils-${BINUTILS_VERSION}.tar.gz
+    cd binutils-${BINUTILS_VERSION}
+    
+    # Configure, compile, and install binutils
+    ./configure --prefix=${BINUTILS_INSTALL_DIR}
+    make -j$(nproc)
+    make install
+    
+    # Clean up
+    cd ..
+    rm -rf binutils-${BINUTILS_VERSION} binutils-${BINUTILS_VERSION}.tar.gz
+    
+    echo "binutils-dev installed successfully in ${BINUTILS_INSTALL_DIR}"
+    BINUTILS_HEADER_DIR=${BINUTILS_INSTALL_DIR}/include
+  fi
 fi
 
 # Add build type condition
@@ -180,10 +195,9 @@ if [ "$LINK_TYPE" == "static" ]; then
     -DLLVM_ENABLE_PROJECTS="clang;openmp;clang-tools-extra" \
     -DLLVM_TARGETS_TO_BUILD=host \
     -DLLVM_EXPERIMENTAL_TARGETS_TO_BUILD="SPIRV" \
-    -DCMAKE_CXX_COMPILER=g++ \
-    -DCMAKE_C_COMPILER=gcc \
     -DLLVM_ENABLE_ASSERTIONS=On \
-    -DLLVM_BINUTILS_INCDIR=${BINUTILS_HEADER_DIR}
+    -DLLVM_BINUTILS_INCDIR=${BINUTILS_HEADER_DIR} \
+    -DCMAKE_CXX_LINK_FLAGS="-Wl,-rpath,${gcc_base_path}/lib64 -L${gcc_base_path}/lib64"
 elif [ "$LINK_TYPE" == "dynamic" ]; then
   cmake ../ \
     -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} \
@@ -195,8 +209,7 @@ elif [ "$LINK_TYPE" == "dynamic" ]; then
     -DLLVM_BUILD_LLVM_DYLIB=ON \
     -DLLVM_PARALLEL_LINK_JOBS=2 \
     -DCMAKE_BUILD_TYPE=Release \
-    -DCMAKE_CXX_COMPILER=g++ \
-    -DCMAKE_C_COMPILER=gcc \
+    -DCMAKE_CXX_LINK_FLAGS="-Wl,-rpath,${gcc_base_path}/lib64 -L${gcc_base_path}/lib64" \
     -DLLVM_BINUTILS_INCDIR=${BINUTILS_HEADER_DIR} \
     -DLLVM_ENABLE_ASSERTIONS=On
 else

--- a/src/CHIPBackend.cc
+++ b/src/CHIPBackend.cc
@@ -177,13 +177,18 @@ void chipstar::AllocationTracker::recordAllocation(
   AllocInfos_.insert(AllocInfo);
 
   if (DevPtr) {
-    assert(!PtrToAllocInfo_.count(DevPtr) &&
-           "Device pointer already recorded!");
-    PtrToAllocInfo_[DevPtr] = AllocInfo;
+    if (!PtrToAllocInfo_.count(DevPtr))
+      PtrToAllocInfo_[DevPtr] = AllocInfo;
+    else
+      CHIPERR_LOG_AND_ABORT("Device pointer already recorded: 0x" +
+                            std::to_string((uintptr_t)DevPtr));
   }
   if (HostPtr) {
-    assert(!PtrToAllocInfo_.count(DevPtr) && "Host pointer already recorded!");
-    PtrToAllocInfo_[HostPtr] = AllocInfo;
+    if (!PtrToAllocInfo_.count(HostPtr))
+      PtrToAllocInfo_[HostPtr] = AllocInfo;
+    else
+      CHIPERR_LOG_AND_ABORT("Host pointer already recorded: 0x" +
+                            std::to_string((uintptr_t)HostPtr));
   }
 
   logDebug("chipstar::AllocationTracker::recordAllocation size: {} HOST {} DEV "
@@ -423,7 +428,7 @@ SPVFuncInfo *chipstar::Module::findFunctionInfo(const std::string &FName) {
 //*************************************************************************************
 chipstar::Kernel::Kernel(std::string HostFName, SPVFuncInfo *FuncInfo)
     : HostFName_(HostFName), FuncInfo_(FuncInfo) {}
-chipstar::Kernel::~Kernel(){};
+chipstar::Kernel::~Kernel() {};
 std::string chipstar::Kernel::getName() { return HostFName_; }
 const void *chipstar::Kernel::getHostPtr() { return HostFPtr_; }
 const void *chipstar::Kernel::getDevPtr() { return DevFPtr_; }
@@ -483,7 +488,7 @@ void *chipstar::ArgSpillBuffer ::allocate(const SPVFuncInfo::Arg &Arg) {
 chipstar::ExecItem::ExecItem(dim3 GridDim, dim3 BlockDim, size_t SharedMem,
                              hipStream_t ChipQueue)
     : SharedMem_(SharedMem), GridDim_(GridDim), BlockDim_(BlockDim),
-      ChipQueue_(static_cast<chipstar::Queue *>(ChipQueue)){};
+      ChipQueue_(static_cast<chipstar::Queue *>(ChipQueue)) {};
 
 dim3 chipstar::ExecItem::getBlock() { return BlockDim_; }
 dim3 chipstar::ExecItem::getGrid() { return GridDim_; }
@@ -1465,7 +1470,7 @@ chipstar::Queue::Queue(chipstar::Device *ChipDevice, chipstar::QueueFlags Flags,
 };
 
 chipstar::Queue::Queue(chipstar::Device *ChipDevice, chipstar::QueueFlags Flags)
-    : Queue(ChipDevice, Flags, 0){};
+    : Queue(ChipDevice, Flags, 0) {};
 
 chipstar::Queue::~Queue() {
   updateLastEvent(nullptr);

--- a/src/CHIPBackend.hh
+++ b/src/CHIPBackend.hh
@@ -2185,6 +2185,13 @@ public:
     return LastEvent_;
   }
 
+  virtual std::shared_ptr<chipstar::Event> getLastEventNoLock() {
+    if (LastEvent_)
+      LastEvent_->isDeletedSanityCheck();
+    return LastEvent_;
+  }
+
+
   /**
    * @brief Construct a new Queue object
    *

--- a/src/CHIPBackend.hh
+++ b/src/CHIPBackend.hh
@@ -269,7 +269,7 @@ class HostAllocFlags {
   unsigned int FlagsRaw_;
 
 public:
-  HostAllocFlags() : FlagsRaw_(hipHostMallocDefault){};
+  HostAllocFlags() : FlagsRaw_(hipHostMallocDefault) {};
   HostAllocFlags(unsigned int FlagsRaw) : FlagsRaw_(FlagsRaw) {
     if (FlagsRaw & hipHostMallocDefault) {
       Default_ = true;
@@ -384,7 +384,7 @@ public:
     Monitor->monitor();
     return 0;
   }
-  virtual void monitor(){};
+  virtual void monitor() {};
 
   void start() {
     logDebug("Starting chipstar::Event Monitor Thread");
@@ -2190,7 +2190,6 @@ public:
       LastEvent_->isDeletedSanityCheck();
     return LastEvent_;
   }
-
 
   /**
    * @brief Construct a new Queue object

--- a/src/CHIPBindings.cc
+++ b/src/CHIPBindings.cc
@@ -2749,8 +2749,7 @@ hipError_t hipEventSynchronize(hipEvent_t Event) {
   NULLCHECK(Event);
   chipstar::Event *ChipEvent = static_cast<chipstar::Event *>(Event);
 
-  if (ChipEvent->isRecordingOrRecorded())
-    ChipEvent->wait();
+  ChipEvent->wait();
 
   RETURN(hipSuccess);
 
@@ -2778,11 +2777,6 @@ hipError_t hipEventElapsedTime(float *Ms, hipEvent_t Start, hipEvent_t Stop) {
     CHIPERR_LOG_AND_THROW("One of the events was not recorded",
                           hipErrorInvalidHandle);
   }
-
-  if (!ChipEventStart->getEventStatus() == EVENT_STATUS_RECORDING)
-    RETURN(hipErrorNotReady);
-  if (!ChipEventStop->getEventStatus() == EVENT_STATUS_RECORDING)
-    RETURN(hipErrorNotReady);
 
   *Ms = ChipEventStart->getElapsedTime(ChipEventStop);
   RETURN(hipSuccess);

--- a/src/CHIPBindings.cc
+++ b/src/CHIPBindings.cc
@@ -2749,7 +2749,8 @@ hipError_t hipEventSynchronize(hipEvent_t Event) {
   NULLCHECK(Event);
   chipstar::Event *ChipEvent = static_cast<chipstar::Event *>(Event);
 
-  ChipEvent->wait();
+  if (ChipEvent->isRecordingOrRecorded())
+    ChipEvent->wait();
 
   RETURN(hipSuccess);
 

--- a/src/CHIPException.hh
+++ b/src/CHIPException.hh
@@ -37,6 +37,10 @@
 #include "CHIPBindingsInternal.hh"
 #include "logging.hh"
 #include <string>
+#include <execinfo.h>
+#include <cxxabi.h>
+#include <dlfcn.h>
+
 class CHIPError {
   std::string Msg_;
   hipError_t Err_;
@@ -61,6 +65,23 @@ public:
 #define CHIPERR_LOG_AND_ABORT(msg)                                             \
   do {                                                                         \
     logError("{} in {}:{}:{}\n", msg, __FILE__, __LINE__, __func__);           \
+    void *callstack[128];                                                      \
+    int frames = backtrace(callstack, 128);                                    \
+    char **strs = backtrace_symbols(callstack, frames);                        \
+    for (int i = 0; i < frames; ++i) {                                         \
+      Dl_info info;                                                            \
+      if (dladdr(callstack[i], &info)) {                                       \
+        char *demangled = NULL;                                                \
+        int status;                                                            \
+        demangled = abi::__cxa_demangle(info.dli_sname, NULL, 0, &status);     \
+        logError("#{} {}", i, demangled ? demangled : info.dli_sname,          \
+                 info.dli_fname);                                              \
+        free(demangled);                                                       \
+      } else {                                                                 \
+        logError("#{} {}", i, strs[i]);                                        \
+      }                                                                        \
+    }                                                                          \
+    free(strs);                                                                \
     std::abort();                                                              \
   } while (0)
 

--- a/src/backend/Level0/CHIPBackendLevel0.cc
+++ b/src/backend/Level0/CHIPBackendLevel0.cc
@@ -468,13 +468,23 @@ float CHIPEventLevel0::getElapsedTime(chipstar::Event *OtherIn) {
   logTrace("CHIPEventLevel0::getElapsedTime()");
   CHIPEventLevel0 *Other = (CHIPEventLevel0 *)OtherIn;
   LOCK(Backend->EventsMtx); // chipstar::Backend::Events_
-  this->updateFinishStatus();
-  Other->updateFinishStatus();
-  if (!this->isFinished() || !Other->isFinished())
-    std::abort();
-  // CHIPERR_LOG_AND_ABORT("One of the events for getElapsedTime() was done
-  // yet",
-  //                       hipErrorNotReady);
+  this->updateFinishStatus(false);
+  Other->updateFinishStatus(false);
+  if (this->getEventStatus() != EVENT_STATUS_RECORDED) {
+    if (Other->getEventStatus() != EVENT_STATUS_RECORDED) {
+      CHIPERR_LOG_AND_THROW(
+          "CHIPEventLevel0::getElapsedTime() neither start nor stop event is recorded",
+          hipErrorNotReady);
+    } else {
+      CHIPERR_LOG_AND_THROW(
+          "CHIPEventLevel0::getElapsedTime() this(start) event is not recorded",
+          hipErrorNotReady);
+    }
+  } else if (Other->getEventStatus() != EVENT_STATUS_RECORDED) {
+    CHIPERR_LOG_AND_THROW(
+        "CHIPEventLevel0::getElapsedTime() other(stop) event is not recorded",
+        hipErrorNotReady);
+  }
 
   uint32_t Started = this->getFinishTime();
   uint32_t Finished = Other->getFinishTime();

--- a/src/backend/Level0/CHIPBackendLevel0.cc
+++ b/src/backend/Level0/CHIPBackendLevel0.cc
@@ -472,9 +472,9 @@ float CHIPEventLevel0::getElapsedTime(chipstar::Event *OtherIn) {
   Other->updateFinishStatus(false);
   if (this->getEventStatus() != EVENT_STATUS_RECORDED) {
     if (Other->getEventStatus() != EVENT_STATUS_RECORDED) {
-      CHIPERR_LOG_AND_THROW(
-          "CHIPEventLevel0::getElapsedTime() neither start nor stop event is recorded",
-          hipErrorNotReady);
+      CHIPERR_LOG_AND_THROW("CHIPEventLevel0::getElapsedTime() neither start "
+                            "nor stop event is recorded",
+                            hipErrorNotReady);
     } else {
       CHIPERR_LOG_AND_THROW(
           "CHIPEventLevel0::getElapsedTime() this(start) event is not recorded",
@@ -1471,10 +1471,13 @@ void CHIPQueueLevel0::finish() {
   if (LastEvent)
     LastEvent->wait();
 
-  zeStatus =
-      zeCommandQueueSynchronize(ZeCmdQ_, ChipEnvVars.getL0EventTimeout());
+  if (zeCmdQOwnership_) {
+    zeStatus =
+        zeCommandQueueSynchronize(ZeCmdQ_, ChipEnvVars.getL0EventTimeout());
   CHIPERR_CHECK_LOG_AND_THROW_TABLE(zeCommandQueueSynchronize,
                                     "zeCommandQueueSynchronize timeout out");
+  }
+
   this->LastEvent_ = nullptr;
   return;
 }

--- a/src/backend/Level0/CHIPBackendLevel0.cc
+++ b/src/backend/Level0/CHIPBackendLevel0.cc
@@ -1474,8 +1474,8 @@ void CHIPQueueLevel0::finish() {
   if (zeCmdQOwnership_) {
     zeStatus =
         zeCommandQueueSynchronize(ZeCmdQ_, ChipEnvVars.getL0EventTimeout());
-  CHIPERR_CHECK_LOG_AND_THROW_TABLE(zeCommandQueueSynchronize,
-                                    "zeCommandQueueSynchronize timeout out");
+    CHIPERR_CHECK_LOG_AND_THROW_TABLE(zeCommandQueueSynchronize,
+                                      "zeCommandQueueSynchronize timeout out");
   }
 
   this->LastEvent_ = nullptr;

--- a/src/backend/Level0/CHIPBackendLevel0.cc
+++ b/src/backend/Level0/CHIPBackendLevel0.cc
@@ -1456,7 +1456,8 @@ CHIPQueueLevel0::memCopyAsyncImpl(void *Dst, const void *Src, size_t Size,
 }
 
 void CHIPQueueLevel0::finish() {
-  auto LastEvent = getLastEvent();
+  LOCK(LastEventMtx); // Queue::LastEvent_
+  auto LastEvent = getLastEventNoLock();
   if (LastEvent)
     LastEvent->wait();
 

--- a/tests/known_failures.yaml
+++ b/tests/known_failures.yaml
@@ -1,6 +1,7 @@
 TOTAL_TESTS: 1397
 ANY:
   ALL:
+    hipStreamSemantics: 'timing dependent, flaky when ctest -j'
     Unit_AnyAll_CompileTest: 'Failed'
     Unit_funnelshift: 'Failed'
     Unit_threadfence_system: 'SEGFAULT'

--- a/tests/known_failures.yaml
+++ b/tests/known_failures.yaml
@@ -551,11 +551,18 @@ salami:
     cuda-matrixMul: '' 
     cuda-bandwidthTest: '' 
   OPENCL_POCL:
-x4\d\d\dc\ds\db0n0: # aurora
+x4\d\d\dc\ds\db0n0: # aurora intel_compute_runtime/release/775.20
   ALL:
   LEVEL0_GPU:
+  	firstTouch: 'Runtime bug'
+  	TestIndirectMappedHostAlloc: '*OutH = 0'
+  	hip_async_binomial: '[2] inf 0x7f800000  5.398513 0x40acc09e'
+  	cuda-lambda: 'OutH == 2'
   OPENCL_CPU:
   OPENCL_GPU:
+    Unit_hipStreamAddCallback_WithDefaultStream: 'timeout'
+	  hip_sycl_interop: 'pi::getPlugin couldn't find plugin -59 (PI_ERROR_INVALID_OPERATION)'
+	  hip_sycl_interop_no_buffers: 'pi::getPlugin couldn't find plugin -59 (PI_ERROR_INVALID_OPERATION)'
   OPENCL_POCL:
 x1921.*b0n0: # sunspot
   ALL:

--- a/tests/known_failures.yaml
+++ b/tests/known_failures.yaml
@@ -551,3 +551,15 @@ salami:
     cuda-matrixMul: '' 
     cuda-bandwidthTest: '' 
   OPENCL_POCL:
+x4\d\d\dc\ds\db0n0: # aurora
+  ALL:
+  LEVEL0_GPU:
+  OPENCL_CPU:
+  OPENCL_GPU:
+  OPENCL_POCL:
+x1921.*b0n0: # sunspot
+  ALL:
+  LEVEL0_GPU:
+  OPENCL_CPU:
+  OPENCL_GPU:
+  OPENCL_POCL:

--- a/tests/known_failures.yaml
+++ b/tests/known_failures.yaml
@@ -484,6 +484,7 @@ salami:
   LEVEL0_GPU:
   OPENCL_CPU:
   OPENCL_GPU:
+    clock: 'timing dependent, flaky'
     hipStreamSemantics: 'TestStreamSemantics_2 (non-blocking stream):  Failed, then timeout'
     Unit_hipGraphAddEventRecordNode_Functional_ElapsedTime: 'Timeout'
     Unit_hipGraphEventRecordNodeSetEvent_SetEventProperty: 'Timeout'

--- a/tests/known_failures.yaml
+++ b/tests/known_failures.yaml
@@ -484,6 +484,7 @@ salami:
   LEVEL0_GPU:
   OPENCL_CPU:
   OPENCL_GPU:
+    hipDynamicShared: 'flaky'
     clock: 'timing dependent, flaky'
     hipStreamSemantics: 'TestStreamSemantics_2 (non-blocking stream):  Failed, then timeout'
     Unit_hipGraphAddEventRecordNode_Functional_ElapsedTime: 'Timeout'

--- a/tests/known_failures.yaml
+++ b/tests/known_failures.yaml
@@ -554,15 +554,15 @@ salami:
 x4\d\d\dc\ds\db0n0: # aurora intel_compute_runtime/release/775.20
   ALL:
   LEVEL0_GPU:
-  	firstTouch: 'Runtime bug'
-  	TestIndirectMappedHostAlloc: '*OutH = 0'
-  	hip_async_binomial: '[2] inf 0x7f800000  5.398513 0x40acc09e'
-  	cuda-lambda: 'OutH == 2'
+    firstTouch: 'Runtime bug'
+    TestIndirectMappedHostAlloc: '*OutH = 0'
+    hip_async_binomial: '[2] inf 0x7f800000  5.398513 0x40acc09e'
+    cuda-lambda: 'OutH == 2'
   OPENCL_CPU:
   OPENCL_GPU:
     Unit_hipStreamAddCallback_WithDefaultStream: 'timeout'
-	  hip_sycl_interop: 'pi::getPlugin couldn't find plugin -59 (PI_ERROR_INVALID_OPERATION)'
-	  hip_sycl_interop_no_buffers: 'pi::getPlugin couldn't find plugin -59 (PI_ERROR_INVALID_OPERATION)'
+    hip_sycl_interop: 'pi::getPlugin couldnt find plugin'
+    hip_sycl_interop_no_buffers: 'pi::getPlugin couldnt find plugin'
   OPENCL_POCL:
 x1921.*b0n0: # sunspot
   ALL:

